driver/normalizer/normalizer.go

```diff
@@ -15,6 +15,141 @@ var Normalize = Transformers([][]Transformer{
 	{Mappings(Normalizers...)},
 }...)
 
+type opArrHasParams struct {
```

Comment:
Too many things mixed here.

I think it should be more like a splitter. Accept two ops: one for a variadic flag, and the second one for a remaining array. On `Check` it will find the variadic modifier and remove it from an array (make sure to clone an array first!). It will then pass the flag to the first sub-op and the remaining array to the second one.

On `Construct` it should do the opposite: get a flag from the first op, an array from the second one, and synthesize a modifier if the flag is set to true.

In general (and we should write it down to docs), custom transforms should be used only on one side of the mapping. Using those kinds of custom transforms for the same vars on both sides indicate that you are doing something wrong. For this case, you can pass two `Var`s as sub-ops and then just use `Var` on the second half of the mapping.

Reply:
Yes, I still don't understand well how they work, better docs and examples would be great, and maybe renaming `Check` and `Construct` to something else, specially `Check` since it doesn't only check, anyway I'll have to change it since I've just discovered that there is another, older way, to pass variable arguments to functions.

Reply:
Yeah, `Check` is a really confusing name, maybe something like `Extract` will be better.